src/search/merge_and_shrink/transition_system.cc

```diff
@@ -28,47 +25,95 @@ ostream &operator<<(ostream &os, const Transition &trans) {
     return os;
 }
 
-// Sorts the given set of transitions and removes duplicates.
-static void normalize_given_transitions(vector<Transition> &transitions) {
-    sort(transitions.begin(), transitions.end());
-    transitions.erase(unique(transitions.begin(), transitions.end()), transitions.end());
+/*
+  Class for representing groups of labels with equivalent transitions in a
+  transition system. See also documentation for TransitionSystem.
+
+  The local label is in a consistent state if label_group and transitions
+  are sorted and unique.
+*/
+LocalLabelInfo::LocalLabelInfo(
+    LabelGroup &&label_group,
+    std::vector<Transition> &&transitions,
+    int cost)
+    : label_group(move(label_group)),
+      transitions(move(transitions)),
+      cost(cost) {
+    assert(utils::is_sorted_unique(this->label_group) &&
+        utils::is_sorted_unique(this->transitions));
 }
 
-TSConstIterator::TSConstIterator(
-    const LabelEquivalenceRelation &label_equivalence_relation,
-    const vector<vector<Transition>> &transitions_by_group_id,
-    bool end)
-    : label_equivalence_relation(label_equivalence_relation),
-      transitions_by_group_id(transitions_by_group_id),
-      current_group_id((end ? label_equivalence_relation.get_size() : 0)) {
-    next_valid_index();
+void LocalLabelInfo::add_label(int label, int label_cost) {
+    label_group.push_back(label);
+    if (label_cost != -1) {
+        cost = min(cost, label_cost);
+    }
+    assert(is_consistent());
 }
 
-void TSConstIterator::next_valid_index() {
-    while (current_group_id < label_equivalence_relation.get_size()
-           && label_equivalence_relation.is_empty_group(current_group_id)) {
-        ++current_group_id;
+void LocalLabelInfo::apply_same_cost_label_mapping(
```

Comment:
This does not more than calling recompute_cost and adding a label; so maybe the name isn't ideal?